### PR TITLE
Add section about network identity.

### PR DESCRIPTION
**Pull request by mpapierski**
### Related links

Needs to wait until casper-network/casper-node#3291 is merged
Follow up for casper-network/casper-node#3268

### Changes

Adds a section about how to configure a network access control to restrict unapproved nodes from joining.

### CODE REVIEW

File: source/docs/casper/workflow/setup-private-network.md [hunk 1]
```diff
@@ -221,6 +221,57 @@ export CHAIN_NAME="private-test"
 
 This testing example will also use an `alice/secret_key.pem` file, a secret key generated through the [keys generation process](/dapp-dev-guide/keys/#creating-accounts-and-keys). Alice is a regular user in this testing example.
 
+### Network access control
+
+With a default configuration each node generates a self-signed certificate to encrypt the peer-to-peer communication. This means any person can join an existing network, and sync with the network, which in private chains may not be allowed.
+
+To restrict access for new nodes to join existing private chain network the node software supports loading signed client certificates by a certificate authority.
```
Comment (ACStoneCL) [inline source/docs/casper/workflow/setup-private-network.md:228]:
```suggestion
To restrict access for new nodes joining an existing private chain network, the node software supports loading signed client certificates by a certificate authority.
```

File: source/docs/casper/workflow/setup-private-network.md [hunk 2]
```diff
@@ -221,6 +221,57 @@ export CHAIN_NAME="private-test"
...
+
+```toml
+[network.identity]
+tls_certificate = "local_node_cert.pem"
+secret_key = "local_node.pem"
+ca_certificate = "ca_cert.pem"
+```
+
+- `tls_certificate` is the certificate signed by a `ca_cert.pem`.
+- `secret_key` refers to a secret key specific that should be unique to a specific node in the network. All peer to peer communication coming from this node will be signed by this key
```
Comment (ACStoneCL) [inline source/docs/casper/workflow/setup-private-network.md:238]:
```suggestion
- `secret_key` refers to a secret key that should be unique to a specific node in the network. All peer-to-peer communication coming from this node will be signed by this key.
```

File: source/docs/casper/workflow/setup-private-network.md [hunk 3]
```diff
@@ -221,6 +221,57 @@ export CHAIN_NAME="private-test"
...
+And then configure the node with following settings:
+
+```toml
+[network.identity]
+tls_certificate = "node_1_cert.pem"
+secret_key = "node_1.pem"
+ca_certificate = "ca_cert.pem"
+```
+
+Every node in the private chain network has to be configured with the same CA certificate, and each `tls_certificate` and `secret_key` pair has to be signed by it. Any node trying to join with a certificate signed by incorrect CA ends up with a following log message:
```
Comment (ACStoneCL) [inline source/docs/casper/workflow/setup-private-network.md:267]:
```suggestion
Every node in the private chain network has to be configured with the same CA certificate, and each `tls_certificate` and `secret_key` pair has to be signed by it. Any node trying to join with a certificate signed by an incorrect CA ends up with the following log message:
```

File: source/docs/casper/workflow/setup-private-network.md [hunk 4]
```diff
@@ -221,6 +221,57 @@ export CHAIN_NAME="private-test"
...
+
+```
+2022-09-01T12:08:53.031417Z DEBUG init:incoming{; peer_addr=127.0.0.1:50998}: [casper_node::components::small_network small_network.rs:501] incoming connection failed early; err=TLS validation error of peer certificate: the certificate is not signed by provided certificate authority
+```
+
+Keep in mind that for security reasons `ca_key.pem` should not be stored securely and never present on each of participating machines.
```
Comment (ACStoneCL) [inline source/docs/casper/workflow/setup-private-network.md:273]:
```suggestion
Keep in mind that for security reasons `ca_key.pem` should not be stored securely and never present on each of participating machines.
```

Should not or should be stored securely?

File: source/docs/casper/workflow/setup-private-network.md (hunk 1 above)
Comment (ipopescu) [reply source/docs/casper/workflow/setup-private-network.md:228]: I've added the CA acronym.

File: source/docs/casper/workflow/setup-private-network.md (hunk 4 above)
Comment (mpapierski) [reply source/docs/casper/workflow/setup-private-network.md]: Good catch! Fixed

**Review by ACStoneCL — APPROVED** [review]

**Pull request merged**